src/applications/disability-benefits/all-claims/utils.jsx

```diff
@@ -263,108 +268,212 @@ export const bankFieldsHaveInput = formData =>
+          options: { addressPath },
+          // pathWithIndex is called in validateMilitaryCity
+          validator: validateMilitaryCity,
+        },
+      ],
+      'ui:errorMessages': {
+        pattern: 'Please fill in a valid city',
+      },
+    },
+    state: {
+      'ui:title': 'State',
+      'ui:required': (formData, index) =>
+        fieldsAreRequired &&
+        _.get(`${pathWithIndex(addressPath, index)}.country`, formData, '') ===
+          USA,
+      'ui:options': {
+        hideIf: (formData, index) =>
+          fieldsAreRequired &&
+          _.get(
+            `${pathWithIndex(addressPath, index)}.country`,
+            formData,
+            '',
+          ) !== USA,
+        updateSchema: updateStates,
+      },
+      'ui:validations': [
+        {
+          options: { addressPath },
+          // pathWithIndex is called in validateMilitaryState
+          validator: validateMilitaryState,
+        },
+      ],
+    },
+    zipCode: {
+      'ui:title': 'Postal code',
       'ui:validations': [validateZIP],
+      'ui:required': (formData, index) =>
+        fieldsAreRequired &&
+        _.get(`${pathWithIndex(addressPath, index)}.country`, formData, '') ===
+          USA,
       'ui:errorMessages': {
         pattern: 'Please enter a valid 5- or 9-digit ZIP code (dashes allowed)',
       },
-    };
-  }
-
-  return locationSchema;
-}
-
-// Could be changed to use generateLocationSchemas
-export function incidentLocationSchemas() {
-  const addressOmitions = [
-    'addressLine1',
-    'addressLine2',
-    'addressLine3',
-    'postalCode',
-    'zipCode',
-  ];
+      'ui:options': {
+        widgetClassNames: 'va-input-medium-large',
+        hideIf: (formData, index) =>
+          _.get(
+            `${pathWithIndex(addressPath, index)}.country`,
+            formData,
+            '',
+          ) !== USA,
+      },
+    },
+  };
+};
 
-  const addressSchemaConfig = addressSchema(fullSchema);
-  const addressUIConfig = omit(addressUI(' '), addressOmitions);
+const ptsdAddressOmitions = [
+  'addressLine1',
+  'addressLine2',
+  'addressLine3',
+  'postalCode',
+  'zipCode',
+];
 
+/**
+ * @param {string} addressPath - The path to the address in the formData
+ */
+export function incidentLocationUISchema(addressPath) {
+  const addressUIConfig = omit(
+    addressUISchema(addressPath, null, false, false),
+    ptsdAddressOmitions,
+  );
   return {
-    addressUI: {
-      ...addressUIConfig,
-      state: {
-        ...addressUIConfig.state,
-        'ui:title': 'State/Province',
-      },
-      additionalDetails: {
-        'ui:title':
-          'Additional details (This could include an address, landmark, military installation, or other location.)',
-        'ui:widget': 'textarea',
-      },
-      'ui:order': ['country', 'state', 'city', 'additionalDetails'],
+    ...addressUIConfig,
+    state: {
+      ...addressUIConfig.state,
+      'ui:title': 'State/Province',
     },
-    addressSchema: {
```

Comment:
The `uiSchema` that this function returned has been pulled out into `incidentLocationUISchema`. This `schema` will be reflected in `vets-json-schema` when [this PR](https://github.com/department-of-veterans-affairs/vets-json-schema/pull/333) gets merged in.